src/main/kotlin/no/nav/familie/ef/sak/api/beregning/BeregningController.kt

```diff
@@ -35,14 +37,28 @@ class BeregningController(private val stegService: StegService,
                 fagsak.hentAktivIdent(),
                 vedtaksdato = LocalDate.now(),
                 behandlingId = behandling.id,
-                andelerTilkjentYtelse = beløpsperioder.map { AndelTilkjentYtelseDTO(beløp = it.beløp.toInt(),
-                                                                      stønadFom = it.fraOgMedDato,
-                                                                      stønadTom = it.tilDato,
-                                                                      kildeBehandlingId = behandling.id,
-                                                                      personIdent = fagsak.hentAktivIdent())
+                andelerTilkjentYtelse = beløpsperioder.map {
+                    AndelTilkjentYtelseDTO(beløp = it.beløp.toInt(),
+                                           stønadFom = it.fraOgMedDato,
+                                           stønadTom = it.tilDato,
+                                           kildeBehandlingId = behandling.id,
+                                           personIdent = fagsak.hentAktivIdent())
                 }
         )
         return Ressurs.success(stegService.håndterBeregnYtelseForStønad(behandling, tilkjentYtelse).id)
     }
 
+    @PostMapping("/{behandlingId}/lagre-vedtak")
+    fun lagreVedtak(@PathVariable behandlingId: UUID, @RequestBody vedtakRequest: VedtakRequest): Ressurs<UUID> {
+        tilgangService.validerTilgangTilBehandling(behandlingId)
+        return Ressurs.success(vedtakService.lagreVedtak(vedtakRequest, behandlingId))
+    }
+
+    @GetMapping("/{behandlingId}/hent-soknad")
+    fun hentSøknadData(@PathVariable behandlingId: UUID): Ressurs<SøknadDataDto> {
```

Comment:
To spørsmål:
1. Burde denne ligge i en annen controller?
2. Noen som har et bedre navn?

Reply:
Vet ikke helt flytet for denne, hadde det vært greit. bare ha `/{behandlingId }` ?

Reply:
SøknadController, kanskje?

Reply:
Det er egentlig ikke grunnlag for beregning, det er jo søknadsdatoer man ønsker å vise frem til saksbehandler. Flyttet de over til SøknadController med /{behandlingId}/datoer siden jeg uansett gjorde litt endringer her.